### PR TITLE
Build with sourcemaps

### PR DESCRIPTION
**Pull request by johnjbarton**
Adds a few percent to the build time.

### CODE REVIEW

**Comment by arv** [comment]
Travis failed.

File: src/node/traceur.js [hunk 1]
```diff
@@ -28,5 +28,8 @@ module.exports = {
   __proto__: traceur,
   get require() {
     return require('./require.js');
+  },
+  get binFilename() {
+  	return filename;
```
Comment (arv) [inline src/node/traceur.js]: wrong indent

File: src/node/System.js [hunk 2]
```diff
@@ -19,9 +19,17 @@ var traceur = require('./traceur.js');
...
 var System = new traceur.runtime.NodeTraceurLoader();
 
+var traceurMap;
+
 require('source-map-support').install({
   retrieveSourceMap: function(filename) {
     var map = System.getSourceMap(filename);
+    if (!map && filename === traceur.binFilename) {
+      if (!traceurMap) {
+        traceurMap = fs.readFileSync(traceur.binFilename + '.map', 'utf8');
```
Comment (arv) [inline src/node/System.js]: Another option might be to make this lazy so it is only read if it is ever used.

File: src/node/traceur.js [hunk 3]
```diff
@@ -28,5 +28,8 @@ module.exports = {
   __proto__: traceur,
   get require() {
     return require('./require.js');
+  },
+  get binFilename() {
```
Comment (arv) [inline src/node/traceur.js]: Not to excited about this name.

Comment (johnjbarton) [inline src/node/traceur.js]: compiledOutputFilename

File: src/node/System.js (hunk 2 above)
Comment (johnjbarton) [inline src/node/System.js]: ? How can this be more lazy than it is?

File: src/node/traceur.js (hunk 1 above)
Comment (johnjbarton) [inline src/node/traceur.js]: done

File: src/node/traceur.js (hunk 3 above)
Comment (arv) [inline src/node/traceur.js]: binaryFilename seems better after all... compile* seems to imply something about the target output file.

File: src/node/System.js (hunk 2 above)
Comment (arv) [inline src/node/System.js]: Good point. I didn't read carefully enough.

**Comment by arv** [comment]
LGTM

**Comment by johnjbarton** [comment]
How can I re-try Travis?  I just want to see if the results vary each time.

**Comment by johnjbarton** [comment]
`selfCompiledFilename`?

**Comment by johnjbarton** [comment]
Browser now has source maps for debugging, let me know if you see issues.

**Pull request closed without merging**